src/main/java/uk/gov/ons/census/casesvc/messaging/ReceiptReceiver.java

```diff
@@ -18,7 +19,7 @@ public ReceiptReceiver(QidReceiptService qidReceiptService) {
     this.qidReceiptService = qidReceiptService;
   }
 
-  @Transactional
+  @Transactional(isolation = Isolation.REPEATABLE_READ)
```

Comment:
This will have a massive performance impact. We need to build some kind of 'before' performance test and take a benchmark before we merge this